src/Chunk.ts

```diff
@@ -669,11 +669,21 @@ export default class Chunk {
 				chunkSourcemapChain,
 				options.sourcemapExcludeSources!
 			);
-			map.sources = map.sources.map(sourcePath =>
-				normalize(
-					options.sourcemapPathTransform ? options.sourcemapPathTransform(sourcePath) : sourcePath
-				)
-			);
+			map.sources = map.sources.map(sourcePath => {
+				const { sourcemapPathTransform } = options;
+
+				if (sourcemapPathTransform) {
+					const newSourcePath = sourcemapPathTransform(sourcePath);
+
+					if (typeof newSourcePath !== 'string') {
+						throw new Error('sourcemapPathTransform function must return a string.');
+					}
+
+					return normalize(newSourcePath);
+				}
+
+				return normalize(sourcePath);
```

Comment:
Maybe we could avoid the duplication of `normalize` by moving it into a separate `.map(normalize)`?
Or either overwrite `sourcePath` (I know, not everyone likes this) or assign it to a mutable variable and instead of an early return just normalize once in the end?

Reply:
I'd go for a separate `.map(normalize)` as it's clear and concise. I've avoided overriding sourcePath (or creating another variable) because it might reduce understandability in this context. I'll update it :)